api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticSearchService.scala

```diff
@@ -24,22 +24,34 @@ class ElasticSearchService @Inject()(@Flag("es.index") index: String,
         get(id).from(s"$index/$itemType")
       }
 
-  def findResults(sortByField: String, limit: Int = 10): Future[RichSearchResponse] =
+  private def executeSearch(searchDefinition: SearchDefinition, limit: Int, from: Int): Future[RichSearchResponse] =
```

Comment:
This is in anticipation of the fact that we’ll likely have more methods on this service, and it would be good not to keep repeating the Elasticsearch logic – but there’s still quite a lot of repetition. Is this a good idea?

In Python, we have a concept of _function decorators_ – you can "wrap" a function, replace it with the wrapped version, and that's where you'd add extra arguments. Looks something like:

```python
def decorator(f):
    def wrapper(*args, **kwargs):
        limit = kwargs.pop('limit')
        from = kwargs.pop('from')
        search_defn = f(*args, **kwargs)
        return execute_search(search_defn, limit=limit, from=from)
    return wrapper
```

and then the syntax is:

```python
@decorator
def find_results(sort_by_field):
    ...

@decorator
def simple_string_query_search(query_string):
    ...
```

and the individual functions just spit out the search definition.

Anything like that in Scala?